CMSgov/design-system
# [NO-TICKET] Update release script to support bumping unchanged betas to full releases

## Summary
See the comments in the code. I need to leave now but wanted to throw this up here for feedback. I have not tested this script yet, but the problem I encountered that this is trying to solve is that when checking out an old beta tag and trying to `yarn release`, the `lerna version` command looks at the repo and says that nothing has changed. In fact, what we want to do is force it to bump and re-release the packages that were released in the beta.

"Retag" might not be a good word for it. I just don't have time at the moment to think about it too deeply but wanted to document my findings and my idea for a fix.

### Fixed
- ~Adding a `--retag` argument when running the release script fixes the problem where when trying to make a full release off of beta release tags lerna's bump tool will not recognize any of the packages as having been changed and will not allow you to perform a version bump.~
- Adding a `--force-publish` argument that allows you to specify which packages will be released in the case where they haven't changed since the last beta (which is most of the time)

## How to test
1. Check out latest beta tag with `git checkout tag/@cmsgov/ds-healthcare-gov@6.4.0-beta.4`
2. Comment out these lines:
    ```
    git push --set-upstream origin $BRANCH
    git push origin --tags
    ```
3.  Must commit`git add --all; git commit -m "testing"` because the release script won't let you move forward with uncommitted changes
4. Run the release script with the new argument ~`yarn release --retag`~ and the names of the packages that are being released `yarn release --force-publish @cmsgov/design-system @cmsgov/design-system-docs @cmsgov/ds-healthcare-gov @cmsgov/ds-medicare-gov`
5. Bump all the versions to non-beta and confirm
6. It should have created all the tags locally (see with `git tag`). You'll have to clean this up locally by deleting the tags with `git tag -d <tag-name>`
